src/worker/vm/extensions/console.ts

```diff
@@ -17,18 +19,30 @@ function consoleFormat(...args: unknown[]): string {
 }
 
 export function createConsole(server: Hub, pluginConfig: PluginConfig): ConsoleExtension {
-    async function consolePersist(type: PluginLogEntryType, ...args: unknown[]): Promise<void> {
+    const consoleBuffer = createBuffer({
+        limit: 100,
+        timeoutSeconds: 1,
+        onFlush: async (logEntries: LogEntryPayload[]) => {
+            await server.db.createPluginLogEntries(pluginConfig, logEntries)
+        },
+    })
```

Comment:
This solution seems very high-level, as it only covers `console` and in a per-plugin fashion – compared to Kafka message queuing, which covers the whole plugin server thread. I think if we do want to go for batching here, it may be cleaner and faster to have `createPluginLogEntry` append to a buffer, which would be flush for the whole the thread. That would avoid the situation where there are also _two_ buffer levels for Kafka-produced `console` logs (initial buffer here, then second buffer in `kafkaProducer.queueMessage`).

Reply:
It is indeed. I originally came at this for writing a general batching system, but decided to start with this as the bulk of logs comes from plugins and I had to work on other things.

Reply:
The system is there to take a batch of entries and insert them so it definitely isn't a major step to write worker-level memory batching. I just took over #431 to get this out quickly